ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1215,30 +1237,106 @@ private String checkOtherDrawsQuantity(String id, Date date, String rowObjectId,
             {
                 overages.add(bloodNextInterval);
             }
+            else if (getThresholdCheck())
+            {
+                //only report about problematic bloods nearing the limit in the current transaction
+                //because allBloods contains everything and we want to distinguish them
+                if (blood1.getIsInTransaction())
+                {
+                    double maxAllowableThreshold = maxAllowable - getThresholdLimit();
+                    if (bloodNextInterval > maxAllowableThreshold)
+                    {
+                        closeToThreshold.add(bloodNextInterval);
+                    }
+                }
+            }
         }
 
-        //always report the most severe overage
-        if (!overages.isEmpty())
+        if (!overages.isEmpty() || !closeToThreshold.isEmpty())
+        {
+            StringBuilder errorMsgBuilder = new StringBuilder();
+
+            //always report the most severe overage
+            if (!overages.isEmpty())
+            {
+                errorMsgBuilder.append("Blood volume of ");
```

Comment:
I appreciate the benefit of StringBuilder over concatenation, but if you're going to do this, i think you can chain them, like:
```
errorMsgBuilder.append(rowQuantity)
                .append(" (")
                .append(overages.descendingSet().iterator().next())
                .append(" over ");
```
or something to that effect.